presto-main/src/test/java/com/facebook/presto/operator/scalar/TestTDigestTruncatedMeanFunctions.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.metadata.FunctionManager;
+import com.facebook.presto.operator.aggregation.AggregationTestUtils;
+import com.facebook.presto.operator.aggregation.InternalAggregationFunction;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.SqlVarbinary;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.facebook.presto.tdigest.TDigest;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import static com.facebook.presto.block.BlockAssertions.createDoubleSequenceBlock;
+import static com.facebook.presto.block.BlockAssertions.createDoublesBlock;
+import static com.facebook.presto.block.BlockAssertions.createRLEBlock;
+import static com.facebook.presto.operator.aggregation.AggregationTestUtils.assertAggregation;
+import static com.facebook.presto.operator.aggregation.TestMergeTDigestFunction.TDIGEST_EQUALITY;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.sql.analyzer.TypeSignatureProvider.fromTypes;
+import static com.facebook.presto.tdigest.TDigest.createTDigest;
+import static java.lang.Double.NaN;
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+import static java.lang.String.format;
+
+public class TestTDigestTruncatedMeanFunctions
```

Comment:
Truncated mean is a scalar function.  Why test it like an aggregation function?  Why not test it similar to e.g. `merge_hll`?

Reply:
The aggregation part looked similar to when we test aggregation functions because we first needed to create a digest from a set of values through aggregation. This is how it was being done in `TestQuantileDigestTruncatedMeanFunctions`. It was basically testing `qdigest_agg` and `truncated_mean` at the same time. However, we don't really need to do this since we already tested `tdigest_agg` separately, so I changed it to create the t-digest without using the Presto engine to aggregate a set of values, and then passing it into the `truncated_mean` function as a Slice (by serializing the t-digest).